### PR TITLE
GG-26852 Allow custom authenticators to use SSL certificates

### PR DESCRIPTION
**Pull request by alamar**

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/processors/odbc/odbc/OdbcConnectionContext.java [hunk 1]
```diff
@@ -117,7 +117,8 @@ public OdbcConnectionContext(GridKernalContext ctx, GridNioSession ses, GridSpin
     }
 
     /** {@inheritDoc} */
-    @Override public void initializeFromHandshake(ClientListenerProtocolVersion ver, BinaryReaderExImpl reader)
+    @Override public void initializeFromHandshake(GridNioSession ses,
```
Comment (isapego) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/odbc/odbc/OdbcConnectionContext.java]: If we pass session instance here then we don't need it a as class field anymore. Same for the JDBC.

Comment (alamar) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/odbc/odbc/OdbcConnectionContext.java]: Will do.

File: modules/core/src/main/java/org/apache/ignite/internal/util/nio/GridNioSessionImpl.java [hunk 2]
```diff
@@ -269,6 +273,22 @@ public void reset0() {
...
 
+    /** */
+    @Override public Certificate[] certificates() {
+        GridSslMeta meta = meta(SSL_META.ordinal());
+
+        if (meta != null) {
+            try {
+                return meta.sslEngine().getSession().getPeerCertificates();
+            }
+            catch (SSLPeerUnverifiedException e) {
```
Comment (ptupitsyn) [inline modules/core/src/main/java/org/apache/ignite/internal/util/nio/GridNioSessionImpl.java]: Not sure, but we probably want to pass that exception along, not swallow it. Let the caller deal with it.

**Review by ptupitsyn — APPROVED** [review]

File: modules/core/src/main/java/org/apache/ignite/internal/util/nio/GridNioSessionImpl.java (hunk 2 above)
Comment (alamar) [reply modules/core/src/main/java/org/apache/ignite/internal/util/nio/GridNioSessionImpl.java]: This is expected exception - can be thrown in perfectly legitimate scenarios.

**Pull request closed without merging**